File: aoc2015/aoc2015-java/src/test/java/de/havox_design/aoc2015/day15/Day15Test.java
package de.havox_design.aoc2015.day15;

import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.params.ParameterizedTest;
import org.junit.jupiter.params.provider.Arguments;
import org.junit.jupiter.params.provider.MethodSource;

import java.util.stream.Stream;

class Day15Test {

    @ParameterizedTest
    @MethodSource("getDataForPart1")
    void testPart1(String fileName, int expectedScore) {
        Assertions.assertEquals(expectedScore, ScienceForHungryPeople.solvePart1(fileName));
    }

    private static Stream<Arguments> getDataForPart1() {
        return Stream.of(
                Arguments.of("de/havox_design/aoc2015/day15/day15Pample.txt", 62842880)
        );
    }

    @ParameterizedTest
    @MethodSource("getDataForPart2")
    void testPart2(String fileName, int expectedScore) {
        Assertions.assertEquals(expectedScore, ScienceForHungryPeople.solvePart2(fileName));
    }

    private static Stream<Arguments> getDataForPart2() {
        return Stream.of(
                Arguments.of("de/havox_design/aoc2015/day15/day15Pample.txt", 57600000)
        );
    }
}

File: aoc2022/aoc2022-kotlin/src/test/kotlin/de/havox_design/aoc2022/day15/Day15Test.kt
package de.havox_design.aoc2022.day15

import de.havox_design.aoc.utils.kotlin.helpers.tests.shouldBe
import org.junit.jupiter.params.ParameterizedTest
import org.junit.jupiter.params.provider.Arguments
import org.junit.jupiter.params.provider.MethodSource
import java.util.stream.Stream

class Day15Test {

    @ParameterizedTest
    @MethodSource("getDataForTestProcessPart1")
    fun testProcessPart1(filename: String, desiredRow: Int, expectedResult: Int) =
        BeaconExclusionZone(filename).processPart1(desiredRow).shouldBe(expectedResult)

    @ParameterizedTest
    @MethodSource("getDataForTestProcessPart2")
    fun testProcessPart2(filename: String, caveSize: Int, expectedResult: Long) =
        BeaconExclusionZone(filename).processPart2(caveSize).shouldBe(expectedResult)

    companion object {
        @JvmStatic
        private fun getDataForTestProcessPart1(): Stream<Arguments> =
            Stream.of(
                Arguments.of("de/havox_design/aoc2022/day15/day15Sample.txt", 10, 26)
            )

        @JvmStatic
        private fun getDataForTestProcessPart2(): Stream<Arguments> =
            Stream.of(
                Arguments.of("de/havox_design/aoc2022/day15/day15Sample.txt", 20, 56000011L)
            )
    }
}
